Native image with liquibase on Windows 10

### Describe the bug

If I try to build native image with Gradle on Windows 10, I receive an error:

```
Error: No instances of java.lang.ProcessEnvironment are allowed in the image heap as this class should be initialized at image runtime. To see how this object got instantiated use --trace-object-instantiation=java.lang.ProcessEnvironment.
Detailed message:
Trace: Object was reached by
        reading field java.util.Collections$UnmodifiableMap.m of
                constant java.util.Collections$UnmodifiableMap@22373044 reached by
        reading field liquibase.configuration.pro.EnvironmentValueProvider.environment of
                constant liquibase.configuration.pro.EnvironmentValueProvider@123a4a79 reached by
        reading field java.util.TreeMap$Entry.key of
                constant java.util.TreeMap$Entry@107eeb72 reached by
        reading field java.util.TreeMap$Entry.left of
                constant java.util.TreeMap$Entry@37d395b8 reached by
        reading field java.util.TreeMap$Entry.left of
                constant java.util.TreeMap$Entry@762604c5 reached by
        reading field java.util.TreeMap.root of
                constant java.util.TreeMap@5832c0c5 reached by
        reading field java.util.TreeSet.m of
                constant java.util.TreeSet@61de108f reached by
        reading field liquibase.configuration.LiquibaseConfiguration.configurationValueProviders of
                constant liquibase.configuration.LiquibaseConfiguration@3d42620b reached by
        reading field java.util.TreeMap$Entry.value of
                constant java.util.TreeMap$Entry@28180a6f reached by
        reading field java.util.TreeMap$Entry.left of
                constant java.util.TreeMap$Entry@566b1eb7 reached by
        reading field java.util.TreeMap$Entry.left of
                constant java.util.TreeMap$Entry@7cd260ba reached by
        reading field java.util.TreeMap.root of
                constant java.util.TreeMap@4472f1ac reached by
        reading field liquibase.util.SmartMap.values of
                constant liquibase.util.SmartMap@6d1336a4 reached by
        reading field liquibase.Scope.values of
                constant liquibase.Scope@71b5e012 reached by
        reading field liquibase.SingletonScopeManager.currentScope of
                constant liquibase.SingletonScopeManager@1ed4c999 reached by
        reading field liquibase.Scope.scopeManager

com.oracle.svm.core.util.UserError$UserException: No instances of java.lang.ProcessEnvironment are allowed in the image heap as this class should be initialized at image runtime. To see how this object got instantiated use --trace-object-instantiation=java.lang.ProcessEnvironment.
Detailed message:
Trace: Object was reached by
        reading field java.util.Collections$UnmodifiableMap.m of
                constant java.util.Collections$UnmodifiableMap@22373044 reached by
        reading field liquibase.configuration.pro.EnvironmentValueProvider.environment of
                constant liquibase.configuration.pro.EnvironmentValueProvider@123a4a79 reached by
        reading field java.util.TreeMap$Entry.key of
                constant java.util.TreeMap$Entry@107eeb72 reached by
        reading field java.util.TreeMap$Entry.left of
                constant java.util.TreeMap$Entry@37d395b8 reached by
        reading field java.util.TreeMap$Entry.left of
                constant java.util.TreeMap$Entry@762604c5 reached by
        reading field java.util.TreeMap.root of
                constant java.util.TreeMap@5832c0c5 reached by
        reading field java.util.TreeSet.m of
                constant java.util.TreeSet@61de108f reached by
        reading field liquibase.configuration.LiquibaseConfiguration.configurationValueProviders of
                constant liquibase.configuration.LiquibaseConfiguration@3d42620b reached by
        reading field java.util.TreeMap$Entry.value of
                constant java.util.TreeMap$Entry@28180a6f reached by
        reading field java.util.TreeMap$Entry.left of
                constant java.util.TreeMap$Entry@566b1eb7 reached by
        reading field java.util.TreeMap$Entry.left of
                constant java.util.TreeMap$Entry@7cd260ba reached by
        reading field java.util.TreeMap.root of
                constant java.util.TreeMap@4472f1ac reached by
        reading field liquibase.util.SmartMap.values of
                constant liquibase.util.SmartMap@6d1336a4 reached by
        reading field liquibase.Scope.values of
                constant liquibase.Scope@71b5e012 reached by
        reading field liquibase.SingletonScopeManager.currentScope of
                constant liquibase.SingletonScopeManager@1ed4c999 reached by
        reading field liquibase.Scope.scopeManager

        at com.oracle.svm.core.util.UserError.abort(UserError.java:87)
        at com.oracle.svm.hosted.FallbackFeature.reportAsFallback(FallbackFeature.java:233)
        at com.oracle.svm.hosted.NativeImageGenerator.runPointsToAnalysis(NativeImageGenerator.java:759)
        at com.oracle.svm.hosted.NativeImageGenerator.doRun(NativeImageGenerator.java:529)
        at com.oracle.svm.hosted.NativeImageGenerator.run(NativeImageGenerator.java:488)
        at com.oracle.svm.hosted.NativeImageGeneratorRunner.buildImage(NativeImageGeneratorRunner.java:403)
        at com.oracle.svm.hosted.NativeImageGeneratorRunner.build(NativeImageGeneratorRunner.java:569)
        at com.oracle.svm.hosted.NativeImageGeneratorRunner.main(NativeImageGeneratorRunner.java:122)
        at com.oracle.svm.hosted.NativeImageGeneratorRunner$JDK9Plus.main(NativeImageGeneratorRunner.java:599)
Caused by: com.oracle.graal.pointsto.constraints.UnsupportedFeatureException: No instances of java.lang.ProcessEnvironment are allowed in the image heap as this class should be initialized at image runtime. To see how this object got instantiated use --trace-object-instantiation=java.lang.ProcessEnvironment.
Detailed message:
Trace: Object was reached by
        reading field java.util.Collections$UnmodifiableMap.m of
                constant java.util.Collections$UnmodifiableMap@22373044 reached by
        reading field liquibase.configuration.pro.EnvironmentValueProvider.environment of
                constant liquibase.configuration.pro.EnvironmentValueProvider@123a4a79 reached by
        reading field java.util.TreeMap$Entry.key of
                constant java.util.TreeMap$Entry@107eeb72 reached by
        reading field java.util.TreeMap$Entry.left of
                constant java.util.TreeMap$Entry@37d395b8 reached by
        reading field java.util.TreeMap$Entry.left of
                constant java.util.TreeMap$Entry@762604c5 reached by
        reading field java.util.TreeMap.root of
                constant java.util.TreeMap@5832c0c5 reached by
        reading field java.util.TreeSet.m of
                constant java.util.TreeSet@61de108f reached by
        reading field liquibase.configuration.LiquibaseConfiguration.configurationValueProviders of
                constant liquibase.configuration.LiquibaseConfiguration@3d42620b reached by
        reading field java.util.TreeMap$Entry.value of
                constant java.util.TreeMap$Entry@28180a6f reached by
        reading field java.util.TreeMap$Entry.left of
                constant java.util.TreeMap$Entry@566b1eb7 reached by
        reading field java.util.TreeMap$Entry.left of
                constant java.util.TreeMap$Entry@7cd260ba reached by
        reading field java.util.TreeMap.root of
                constant java.util.TreeMap@4472f1ac reached by
        reading field liquibase.util.SmartMap.values of
                constant liquibase.util.SmartMap@6d1336a4 reached by
        reading field liquibase.Scope.values of
                constant liquibase.Scope@71b5e012 reached by
        reading field liquibase.SingletonScopeManager.currentScope of
                constant liquibase.SingletonScopeManager@1ed4c999 reached by
        reading field liquibase.Scope.scopeManager

        at com.oracle.graal.pointsto.constraints.UnsupportedFeatures.report(UnsupportedFeatures.java:126)
        at com.oracle.svm.hosted.NativeImageGenerator.runPointsToAnalysis(NativeImageGenerator.java:756)
        ... 6 more
Error: Image build request failed with exit status 1
```

### Expected behavior

_No response_

### Actual behavior

_No response_

### How to Reproduce?

Reproducer: io.quarkus:quarkus-liquibase:2.5.4.Final

Steps to reproduce the behavior:
gradlew build -Dquarkus.package.type=native

### Output of `uname -a` or `ver`

Microsoft Windows [Version 10.0.19042.1415]

### Output of `java -version`

openjdk version "11.0.13" 2021-10-19 OpenJDK Runtime Environment GraalVM CE 21.3.0 (build 11.0.13+7-jvmci-21.3-b05) OpenJDK 64-Bit Server VM GraalVM CE 21.3.0 (build 11.0.13+7-jvmci-21.3-b05, mixed mode, sharing)

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

2.5.4.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

 ------------------------------------------------------------ Gradle 7.2 ------------------------------------------------------------  Build time:   2021-08-17 09:59:03 UTC Revision:     a773786b58bb28710e3dc96c4d1a7063628952ad  Kotlin:       1.5.21 Groovy:       3.0.8 Ant:          Apache Ant(TM) version 1.10.9 compiled on September 27 2020 JVM:          11.0.13 (GraalVM Community 11.0.13+7-jvmci-21.3-b05) OS:           Windows 10 10.0 amd64

### Additional information

_No response_

@geoand 
This is example: https://github.com/dnfesenk/liquibase-issue

```
c:\dfesenko\liquibase-issue>gradlew build -Dquarkus.package.type=native

> Task :quarkusBuild
building quarkus jar

Uber JAR strategy is used for native image source JAR generation on Windows. This is done for the time being to work around a current GraalVM limitation on Windows concerning the maximum command length (see https://github.com/oracle/graal/issues/2387).
Dependencies with duplicate files detected. The dependencies [org.jetbrains.kotlin:kotlin-reflect::jar:1.5.31[paths: C:\Users\fesen\.gradle\caches\modules-2\files-2.1\org.jetbrains.kotlin\kotlin-reflect\1.5.31\1523fcd842a47da0820cea772b19c51056fec8a9\kotlin-reflect-1.5.31.jar;], org.jetbrains.kotlin:kotlin-stdlib-jdk7::jar:1.5.31[paths: C:\Users\fesen\.gradle\caches\modules-2\files-2.1\org.jetbrains.kotlin\kotlin-stdlib-jdk7\1.5.31\77e0f2568912e45d26c31fd417a332458508acdf\kotlin-stdlib-jdk7-1.5.31.jar;], org.jetbrains.kotlin:kotlin-stdlib-jdk8::jar:1.5.31[paths: C:\Users\fesen\.gradle\caches\modules-2\files-2.1\org.jetbrains.kotlin\kotlin-stdlib-jdk8\1.5.31\ff5d99aecd328872494e8921b72bf6e3af97af3e\kotlin-stdlib-jdk8-1.5.31.jar;], org.jetbrains.kotlin:kotlin-stdlib::jar:1.5.31[paths: C:\Users\fesen\.gradle\caches\modules-2\files-2.1\org.jetbrains.kotlin\kotlin-stdlib\1.5.31\6628d61d0f5603568e72d2d5915d2c034b4f1c55\kotlin-stdlib-1.5.31.jar;]] contain duplicate files, e.g. META-INF/versions/9/module-info.class
[liquibase-issue-1.0-SNAPSHOT-runner:8988]    classlist:   6,835.59 ms,  0.94 GB
[liquibase-issue-1.0-SNAPSHOT-runner:8988]        (cap):  20,860.66 ms,  0.94 GB
[liquibase-issue-1.0-SNAPSHOT-runner:8988]        setup:  23,526.67 ms,  0.93 GB
The bundle named: messages, has not been found. If the bundle is part of a module, verify the bundle name is a fully qualified class name. Otherwise verify the bundle path is accessible in the classpath.
18:46:48,114 INFO  [org.jbo.threads] JBoss Threads version 3.4.2.Final
[liquibase-issue-1.0-SNAPSHOT-runner:8988]     (clinit):   1,021.04 ms,  5.93 GB
[liquibase-issue-1.0-SNAPSHOT-runner:8988]   (typeflow):   9,698.04 ms,  5.93 GB
[liquibase-issue-1.0-SNAPSHOT-runner:8988]    (objects):  54,350.85 ms,  5.93 GB
[liquibase-issue-1.0-SNAPSHOT-runner:8988]   (features):   7,176.77 ms,  5.93 GB
[liquibase-issue-1.0-SNAPSHOT-runner:8988]     analysis:  74,313.94 ms,  5.93 GB
Error: No instances of java.lang.ProcessEnvironment are allowed in the image heap as this class should be initialized at image runtime. To see how this object got instantiated use --trace-object-instantiation=java.lang.ProcessEnvironment.
Detailed message:
Trace: Object was reached by
        reading field java.util.Collections$UnmodifiableMap.m of
                constant java.util.Collections$UnmodifiableMap@1d1836f0 reached by
        reading field liquibase.configuration.pro.EnvironmentValueProvider.environment of
                constant liquibase.configuration.pro.EnvironmentValueProvider@3b0ffdfe reached by
        reading field java.util.TreeMap$Entry.key of
                constant java.util.TreeMap$Entry@66baa921 reached by
        reading field java.util.TreeMap$Entry.left of
                constant java.util.TreeMap$Entry@396d912e reached by
        reading field java.util.TreeMap$Entry.left of
                constant java.util.TreeMap$Entry@5ec1a615 reached by
        reading field java.util.TreeMap.root of
                constant java.util.TreeMap@46b3ea54 reached by
        reading field java.util.TreeSet.m of
                constant java.util.TreeSet@26ef711d reached by
        reading field liquibase.configuration.LiquibaseConfiguration.configurationValueProviders of
                constant liquibase.configuration.LiquibaseConfiguration@2fe0eb3f reached by
        reading field java.util.TreeMap$Entry.value of
                constant java.util.TreeMap$Entry@e486d30 reached by
        reading field java.util.TreeMap$Entry.left of
                constant java.util.TreeMap$Entry@78d5ade9 reached by
        reading field java.util.TreeMap$Entry.left of
                constant java.util.TreeMap$Entry@19c7e06 reached by
        reading field java.util.TreeMap.root of
                constant java.util.TreeMap@7848d7c5 reached by
        reading field liquibase.util.SmartMap.values of
                constant liquibase.util.SmartMap@67155470 reached by
        reading field liquibase.Scope.values of
                constant liquibase.Scope@7e0951ec reached by
        reading field liquibase.SingletonScopeManager.currentScope of
                constant liquibase.SingletonScopeManager@1945ed70 reached by
        reading field liquibase.Scope.scopeManager

com.oracle.svm.core.util.UserError$UserException: No instances of java.lang.ProcessEnvironment are allowed in the image heap as this class should be initialized at image runtime. To see how this object got instantiated use --trace-object-instantiation=java.lang.ProcessEnvironment.
Detailed message:
Trace: Object was reached by
        reading field java.util.Collections$UnmodifiableMap.m of
                constant java.util.Collections$UnmodifiableMap@1d1836f0 reached by
        reading field liquibase.configuration.pro.EnvironmentValueProvider.environment of
                constant liquibase.configuration.pro.EnvironmentValueProvider@3b0ffdfe reached by
        reading field java.util.TreeMap$Entry.key of
                constant java.util.TreeMap$Entry@66baa921 reached by
[liquibase-issue-1.0-SNAPSHOT-runner:8988]      [total]: 105,661.38 ms,  5.93 GB
        reading field java.util.TreeMap$Entry.left of
                constant java.util.TreeMap$Entry@396d912e reached by
        reading field java.util.TreeMap$Entry.left of
                constant java.util.TreeMap$Entry@5ec1a615 reached by
        reading field java.util.TreeMap.root of
                constant java.util.TreeMap@46b3ea54 reached by
        reading field java.util.TreeSet.m of
                constant java.util.TreeSet@26ef711d reached by
        reading field liquibase.configuration.LiquibaseConfiguration.configurationValueProviders of
                constant liquibase.configuration.LiquibaseConfiguration@2fe0eb3f reached by
        reading field java.util.TreeMap$Entry.value of
                constant java.util.TreeMap$Entry@e486d30 reached by
        reading field java.util.TreeMap$Entry.left of
                constant java.util.TreeMap$Entry@78d5ade9 reached by
        reading field java.util.TreeMap$Entry.left of
                constant java.util.TreeMap$Entry@19c7e06 reached by
        reading field java.util.TreeMap.root of
                constant java.util.TreeMap@7848d7c5 reached by
        reading field liquibase.util.SmartMap.values of
                constant liquibase.util.SmartMap@67155470 reached by
        reading field liquibase.Scope.values of
                constant liquibase.Scope@7e0951ec reached by
        reading field liquibase.SingletonScopeManager.currentScope of
                constant liquibase.SingletonScopeManager@1945ed70 reached by
        reading field liquibase.Scope.scopeManager

        at com.oracle.svm.core.util.UserError.abort(UserError.java:87)
        at com.oracle.svm.hosted.FallbackFeature.reportAsFallback(FallbackFeature.java:233)
        at com.oracle.svm.hosted.NativeImageGenerator.runPointsToAnalysis(NativeImageGenerator.java:759)
        at com.oracle.svm.hosted.NativeImageGenerator.doRun(NativeImageGenerator.java:529)
        at com.oracle.svm.hosted.NativeImageGenerator.run(NativeImageGenerator.java:488)
        at com.oracle.svm.hosted.NativeImageGeneratorRunner.buildImage(NativeImageGeneratorRunner.java:403)
        at com.oracle.svm.hosted.NativeImageGeneratorRunner.build(NativeImageGeneratorRunner.java:569)
        at com.oracle.svm.hosted.NativeImageGeneratorRunner.main(NativeImageGeneratorRunner.java:122)
        at com.oracle.svm.hosted.NativeImageGeneratorRunner$JDK9Plus.main(NativeImageGeneratorRunner.java:599)
Caused by: com.oracle.graal.pointsto.constraints.UnsupportedFeatureException: No instances of java.lang.ProcessEnvironment are allowed in the image heap as this class should be initialized at image runtime. To see how this object got instantiated use --trace-object-instantiation=java.lang.ProcessEnvironment.
Detailed message:
# Printing build artifacts to: C:\dfesenko\liquibase-issue\build\liquibase-issue-1.0-SNAPSHOT-native-image-source-jar\liquibase-issue-1.0-SNAPSHOT-runner.build_artifacts.txt
Trace: Object was reached by
        reading field java.util.Collections$UnmodifiableMap.m of
                constant java.util.Collections$UnmodifiableMap@1d1836f0 reached by
        reading field liquibase.configuration.pro.EnvironmentValueProvider.environment of
                constant liquibase.configuration.pro.EnvironmentValueProvider@3b0ffdfe reached by
        reading field java.util.TreeMap$Entry.key of
                constant java.util.TreeMap$Entry@66baa921 reached by
        reading field java.util.TreeMap$Entry.left of
                constant java.util.TreeMap$Entry@396d912e reached by
        reading field java.util.TreeMap$Entry.left of
                constant java.util.TreeMap$Entry@5ec1a615 reached by
        reading field java.util.TreeMap.root of
                constant java.util.TreeMap@46b3ea54 reached by
        reading field java.util.TreeSet.m of
                constant java.util.TreeSet@26ef711d reached by
        reading field liquibase.configuration.LiquibaseConfiguration.configurationValueProviders of
                constant liquibase.configuration.LiquibaseConfiguration@2fe0eb3f reached by
        reading field java.util.TreeMap$Entry.value of
                constant java.util.TreeMap$Entry@e486d30 reached by
        reading field java.util.TreeMap$Entry.left of
                constant java.util.TreeMap$Entry@78d5ade9 reached by
        reading field java.util.TreeMap$Entry.left of
                constant java.util.TreeMap$Entry@19c7e06 reached by
        reading field java.util.TreeMap.root of
                constant java.util.TreeMap@7848d7c5 reached by
        reading field liquibase.util.SmartMap.values of
                constant liquibase.util.SmartMap@67155470 reached by
        reading field liquibase.Scope.values of
                constant liquibase.Scope@7e0951ec reached by
        reading field liquibase.SingletonScopeManager.currentScope of
                constant liquibase.SingletonScopeManager@1945ed70 reached by
        reading field liquibase.Scope.scopeManager

        at com.oracle.graal.pointsto.constraints.UnsupportedFeatures.report(UnsupportedFeatures.java:126)
        at com.oracle.svm.hosted.NativeImageGenerator.runPointsToAnalysis(NativeImageGenerator.java:756)
        ... 6 more
Error: Image build request failed with exit status 1

> Task :quarkusBuild FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':quarkusBuild'.
> io.quarkus.builder.BuildException: Build failure: Build failed due to errors
        [error]: Build step io.quarkus.deployment.pkg.steps.NativeImageBuildStep#build threw an exception: java.lang.RuntimeException: Failed to build native image
        at io.quarkus.deployment.pkg.steps.NativeImageBuildStep.build(NativeImageBuildStep.java:239)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:887)
        at io.quarkus.builder.BuildContext.run(BuildContext.java:277)
        at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
        at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1478)
        at java.base/java.lang.Thread.run(Thread.java:829)
        at org.jboss.threads.JBossThread.run(JBossThread.java:501)
  Caused by: java.lang.RuntimeException: Image generation failed. Exit code: 1
        at io.quarkus.deployment.pkg.steps.NativeImageBuildStep.imageGenerationFailed(NativeImageBuildStep.java:375)
        at io.quarkus.deployment.pkg.steps.NativeImageBuildStep.build(NativeImageBuildStep.java:218)
        ... 11 more


* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 2m 5s
7 actionable tasks: 1 executed, 6 up-to-date
```

--- a/extensions/liquibase/runtime/src/main/java/io/quarkus/liquibase/runtime/graal/SubstituteEnvironmentValueProvider.java
+++ b/extensions/liquibase/runtime/src/main/java/io/quarkus/liquibase/runtime/graal/SubstituteEnvironmentValueProvider.java
@@ -1,0 +1,20 @@
+package io.quarkus.liquibase.runtime.graal;
+
+import java.util.Map;
+
+import com.oracle.svm.core.annotate.Delete;
+import com.oracle.svm.core.annotate.Substitute;
+import com.oracle.svm.core.annotate.TargetClass;
+
+@TargetClass(className = "liquibase.configuration.pro.EnvironmentValueProvider")
+final class SubstituteEnvironmentValueProvider {
+
+    @Delete
+    private Map<String, String> environment;
+
+    @Substitute
+    protected Map<?, ?> getMap() {
+        return System.getenv();
+    }
+
+}

